compiler/parser/src/lexer.rs

```diff
@@ -495,6 +459,9 @@ where
 
     fn lex_string(&mut self, kind: StringKind) -> LexResult {
         let start_pos = self.get_pos();
+        for _ in 0..kind.to_string().len() {
```

Comment:
we can also avoid this allocation by adding a `len`-like method on `StringKind` that returns the correct amount based on the kind.

Reply:
Agreed. I'll add a `len` method.

Reply:
Added a `prefix_len` method.